projects/samples/contests/robocup/controllers/referee/referee.py

```diff
@@ -2418,7 +2422,10 @@ def main_loop(self):
                 if not self.game.penalty_shootout:
                     ball_holding = self.check_ball_holding()       # check for ball holding fouls
                     if ball_holding:
-                        self.interruption('FREEKICK', ball_holding, self.game.ball_position)
+                        if self.is_game_interruption():
+                            self.game_interruption_touched(ball_holding)
+                        else:
+                            self.interruption('FREEKICK', ball_holding, self.game.ball_position)
```

Comment:
Not sure if I understand this right. Should we *always* call game_interruption_touched() when is_game_interruption() holds, and otherwise interruption()? If yes, I'd assert is_game_interruption() in game_interruption_touched() and assert `not is_game_interruption()` in interruption(). Also, maybe it makes sense to create a wrapper function around for this:

```
                        if self.is_game_interruption():
                            self.game_interruption_touched(ball_holding)
                        else:
                            self.interruption('FREEKICK', ball_holding, self.game.ball_position)
```

and replace all calls to game_interruption_touched() and interruption() with that wrapper function?

Reply:
Yes, that makes sense to me.